### PR TITLE
add check + error message for pre 1.1.0 kv namespace format

### PR DESCRIPTION
**Pull request by xortive**

### CODE REVIEW

**Comment by xortive** [comment]
💁‍  the new format looks like this 💁‍

[[kv-namespaces]]
binding = "FOO"
id = "0f2ac74b498b48028cb68387c421e279"

Error: ⚠️  Your project config contains the old, undocumented kv-namespace format ⚠️

File: src/settings/project/mod.rs [hunk 1]
```diff
@@ -83,6 +84,26 @@ fn get_project_config(config_path: &Path) -> Result<Project, failure::Error> {
...
     s.merge(Environment::with_prefix("CF"))?;
 
+    // check for pre 1.1.0 KV namespace format
+    let kv_namespaces: Result<Vec<config::Value>, config::ConfigError> = s.get("kv-namespaces");
+
+    if let Ok(values) = kv_namespaces {
+        let old_format = values.iter().any(|val| val.clone().into_str().is_ok());
+
+        if old_format {
+            message::warn("As of 1.1.0 the kv-namespaces format was stabilized");
```
Comment (ashleymichal) [inline src/settings/project/mod.rs]:
```suggestion
            message::warn("As of 1.1.0 the kv-namespaces format has been stabilized");
```

File: src/settings/project/mod.rs [hunk 2]
```diff
@@ -83,6 +84,26 @@ fn get_project_config(config_path: &Path) -> Result<Project, failure::Error> {
...
+            message::info("Please add a section like this in your wrangler.toml for each KV Namespace you wish to bind:");
+
+            let fmt_demo = r#"
+[[kv-namespaces]]
+binding = "BINDING_NAME"
+id = "0f2ac74b498b48028cb68387c421e279"
+            "#;
+
+            println!("{}", fmt_demo);
+        }
```
Comment (ashleymichal) [inline src/settings/project/mod.rs]: should this return early?

Comment (xortive) [reply src/settings/project/mod.rs]: nope, I want it to fall through to reuse the normal project error, which with the context about the changed binding format is informative

**Comment by xortive** [comment]
As of commit 08e6495, this will print:
⚠️  As of 1.1.0 the kv-namespaces format was stabilized ⚠️
💁‍  Please add a section like this in your wrangler.toml for each KV Namespace you wish to bind: 💁‍

[[kv-namespaces]]
binding = "BINDING_NAME"
id = "0f2ac74b498b48028cb68387c421e279"

Error: ⚠️  Your project config has an error, check your `wrangler.toml`: invalid type: string "\<first binding in list\>", expected struct KvNamespace

**Comment by xortive** [comment]
going to add a line to replace the id with the one of your namespace

File: src/settings/project/mod.rs (hunk 2 above)
Comment (ashleymichal) [reply src/settings/project/mod.rs]: reading the comment below, i don't think the normal project error necessarily gives any more context, in fact it's repetitive and therefore a little confusing to read.

File: src/settings/project/mod.rs [hunk 3]
```diff
@@ -100,10 +100,13 @@ binding = "BINDING_NAME"
 id = "0f2ac74b498b48028cb68387c421e279"
             "#;
 
-            message::info("binding is the variable name you wish to bind the namespace to in your script");
-            message::info("id is the namespace_id assigned to your kv namespace upon creation. e.g. (per namespace)")
+            message::info(
+                "binding is the variable name you wish to bind the namespace to in your script.",
```
Comment (xortive) [inline src/settings/project/mod.rs]: cargo fmt, whyyyyyy

**Comment by xortive** [comment]
as of 27f04be

⚠️  As of 1.1.0 the kv-namespaces format has been stabilized ⚠️
💁‍  Please add a section like this in your wrangler.toml for each KV Namespace you wish to bind: 💁‍

[[kv-namespaces]]
binding = "BINDING_NAME"
id = "0f2ac74b498b48028cb68387c421e279"

# binding is the variable name you wish to bind the namespace to in your script.
# id is the namespace_id assigned to your kv namespace upon creation. e.g. (per namespace)

Error: ⚠️  Your project config has an error, check your `wrangler.toml`: invalid type: string "test", expected struct KvNamespace

**Comment by signalnerve** [comment]
@xortive can you wrap the output in your comment in a code block?

**Comment by xortive** [comment]
As of e5b417a

```
⚠️  As of 1.1.0 the kv-namespaces format has been stabilized ⚠️
💁‍  Please add a section like this in your `wrangler.toml` for each KV Namespace you wish to bind: 💁‍

[[kv-namespaces]]
binding = "BINDING_NAME"
id = "0f2ac74b498b48028cb68387c421e279"

# binding is the variable name you wish to bind the namespace to in your script.
# id is the namespace_id assigned to your kv namespace upon creation. e.g. (per namespace)

Error: ⚠️  Your project config has an error ⚠️
```

File: src/settings/project/mod.rs (hunk 2 above)
Comment (xortive) [reply src/settings/project/mod.rs]: changed the message but I personally prefer the fall-through one as it explains the exact error in gory detail as well as how to fix it

**Comment by ashleygwilliams** [comment]
@xortive you've got tests failing :( need to run cargo fmt

**Pull request merged**